bcipy/helpers/task.py

```diff
@@ -370,186 +373,159 @@ def get_key_press(
     return None
 
 
+def data_reshaper(input_data_type: InputDataType, *args, **kwargs):
+    if input_data_type == InputDataType.TRIAL:
+        return trial_reshaper(*args, **kwargs)
+    elif input_data_type == InputDataType.INQUIRY:
+        return inquiry_reshaper(*args, **kwargs)
+    else:
+        raise NotImplementedError(f"Unsupported input data type: {input_data_type}")
+
```

Comment:
Rather than dispatching to a function based on an enum value, it seems like this would be well-suited to an object-oriented design using something like a`Reshaper` parent class for the interface with `TrialReshaper` and `InquiryReshaper` subclasses. These could each override a `reshape` method or you could use the `__call__` method like you did for the filters. Using classes would also allow you to use more meaningful type hints.

Reply:
Would be happy to restructure, but I'm not sure I quite understand - for example in `copy_phrase_wrapper`, here's what I'm imagining based on what you described.
Instead of calling a function  `data_reshaper`, it could have something like:

```python
if ...:
    reshaper = TrialReshaper(...)
else:
    reshaper = InquiryReshaper(...)

x, y = reshaper(data)
```

and in `bcipy/helpers/task.py`, something like:

```python
class Reshaper(ABC):
    @abstractmethod
    def __call__(self):
        ...

class TrialReshaper(Reshaper):
    def __call__(self, trial_target_info, timing_info, eeg_data):
        # implementation
```

The issue is this still basically seems to rely on that `if/else` statement happening somewhere.

Another approach might be to store this on the signal model itself, but I think we had decided against this 
(I think the rationale was that we don't necessarily have a signal model every time we want to reshape data, for example in `bcipy/tasks/rsvp/calibration/inter_inquiry_feedback_calibration.py`)

Reply:
Yes, this is what I had in mind, storing the appropriate class with the signal model, rather than the enum. Rather than calling:

`x, y = data_reshaper(input_data_type=model.input_data_type, ...)`

It would be something like:

```
x, y = model.reshape(trial_target_info=t_t_i,
                     timing_info=t_i,
                     eeg_data=data,
                     fs=fs,
                     trials_per_inquiry=parameters.get('stim_length'),
                     offset=offset,
                     channel_map=channel_map,
                     trial_length=trial_length)
```
or 

`x, y = model.reshaper.reshape(...)`

If we don't have the signal model we just use the appropriate reshaper directly.